rocAL/rocAL_pybind/rocal_pybind.cpp

```diff
@@ -316,6 +332,162 @@ namespace rocal{
+                R"code(
+                Returns a tensor buffer's width.
+                )code"
+            )
+            .def(
+                "batch_size",
+                [](rocalTensor &output_tensor) {
+                    return output_tensor.info().dims().at(0);
+                },
+                R"code(
+                Returns a tensor batch size.
+                )code"
+            )
+            .def("layout", [](rocalTensor &output_tensor) {
+                return rocALToPybindLayout[(int)output_tensor.info().layout()];
+            },
+                R"code(
+                Returns layout of tensor.
+                )code"
+            )
+            .def("dtype", [](rocalTensor &output_tensor) {
+                return rocALToPybindOutputDtype[(int)output_tensor.info().data_type()];
+            },
+                R"code(
+                Returns dtype of tensor.
+                )code"
+            )
+            .def("torch_dtype", [](rocalTensor &output_tensor) {
+                return "torch." + rocALToPybindOutputDtype[(int)output_tensor.info().data_type()];
+            },
+                R"code(
+                Returns dtype of torch tensor.
+                )code"
+            )
+            .def("numpy_dtype", [](rocalTensor &output_tensor) {
+                return "np." + rocALToPybindOutputDtype[(int)output_tensor.info().data_type()];
+            },
+                R"code(
+                Returns dtype of numpy array.
+                )code"
+            )
+            .def("cupy_dtype", [](rocalTensor &output_tensor) {
+                return "cp." + rocALToPybindOutputDtype[(int)output_tensor.info().data_type()];
+            },
+                R"code(
+                Returns dtype of cupy array.
+                )code"
+            )
+            .def("dims", [](rocalTensor &output_tensor) {
+                return output_tensor.info().dims();
+            },
+                R"code(
+                Returns dims of tensor.
+                )code"
+            )
+            m.def("dimensions", []=(rocalTensor &output_tensor) {
+            py::list list;
+            for (uint i = 0; i < output_tensor.info().dims(); i++)
+                list.append(output_tensor.info().dims().at(i));
+            return list; 
+            });
+            .def(
+            "copy_data", [](rocalTensor &output_tensor, py::object p) {
+                auto ptr = ctypes_void_ptr(p);
+                output_tensor.copy_data(ptr);
+            },
+                R"code(
+                Copies the ring buffer data to python buffer pointers.
+                )code"
+            )
+            .def(
+                "at",
+                [](rocalTensor &output_tensor, uint idx) {
+                    const auto& info = output_tensor.info();
+                    uint idx_stride = 1;
+                    for (uint i = 0; i < info.num_of_dims() - 1 ; i++) idx_stride*=output_tensor.info().dims().at(i+1);
```

Comment:
Can we not calculate this stride in a single line?
Basically the stride from C++ will have H*W*C*sizeof(type) at one of it's elements can we not make use of that
Please avoid the for loop here